lib/connection/pool.js

```diff
@@ -277,6 +277,8 @@ function attemptReconnect(self) {
     // If we have failure schedule a retry
     function _connectionFailureHandler(self, event) {
       return function() {
+        if (this._connectionFailHandled) return;
```

Comment:
not sure if we can use `this.destroyed` to replace `this._connectionFailHandled `